cosinnus/views/group.py

```diff
@@ -465,6 +474,118 @@ def get_context_data(self, **kwargs):
 group_detail_api = GroupDetailView.as_view(is_ajax_request_url=True)
 
 
+class WorkshopParticipantsView(SamePortalGroupMixin, RequireReadMixin, DetailView):
+
+
+    template_name = 'cosinnus/group/workshop_participants.html'
+
+    def get_object(self, queryset=None):
+        return self.group
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['form'] = CosinusWorkshopParticipantCSVImportForm()
+        return context
+
+    def post(self, request, *args, **kwargs):
+
+        form = CosinusWorkshopParticipantCSVImportForm(request.POST, request.FILES)
+        if form.is_valid():
+            csv_file = request.FILES['participants']
+            decoded_file = csv_file.read().decode('utf-8')
+            header, accounts = self.process_csv(decoded_file)
+
+            filename = '{}_participants.csv'.format(self.group.slug)
+
+            response = HttpResponse(content_type='text/csv')
+            response['Content-Disposition'] = 'attachment; filename="{}"'.format(filename)
+
+            writer = csv.writer(response)
+            writer.writerow(header)
+            for account in accounts:
+                writer.writerow(account)
+
+            return response
+
+    def process_csv(self, file):
+
+        io_string = io.StringIO(file)
+        reader =  csv.reader(io_string, delimiter=';', quotechar='|')
+        header = next(reader, None)
+        groups_list = self.get_groups_from_header(header)
+        accounts_list = []
+        for row in reader:
+            account = self.create_account(row, groups_list)
+            accounts_list.append(account)
+
+        return header + ['email', 'password'], accounts_list
+
+
+    def get_groups_from_header(self, header):
+        groups = []
+        for entry in header:
+            if entry:
+                try:
+                    group = CosinnusGroup.objects.get(parent=self.get_object(),
+                                                      type=CosinnusGroup.TYPE_PROJECT,
+                                                      slug=entry)
+                    groups.append(group)
+                except ObjectDoesNotExist:
+                    groups.append('')
+            else:
+                groups.append('')
+        return groups
+
+    def create_account(self, data, groups):
+
+        username = data[0]
+        email = '{}--{}-nr@wechange.de'.format(str(self.get_object().id), username)
+        pwd = get_random_string()
+
+        user_data = {
+            'username': username,
+            'password1': pwd,
+            'password2': pwd
+        }
+
+        form = UserCreationForm(user_data)
+        if form.is_valid():
+            user = form.save()
+        else:
+            return False
+
+        user.email = email
+        user.username = user.id
+        user.backend = 'cosinnus.backends.EmailAuthBackend'
```

Comment:
@saschan I found this line here: https://github.com/wechange-eg/cosinnus-core/blob/master/cosinnus/utils/user.py#L202

I this necessary?